sqldelight-compiler/dialect/src/main/kotlin/app/cash/sqldelight/dialect/api/TypeResolver.kt

```diff
@@ -68,12 +77,17 @@ fun TypeResolver.encapsulatingType(
     error("The Kotlin type of the argument cannot be inferred, use CAST instead.")
   }
 
-  val type = typeOrder.last { it in sqlTypes }
+  val isTypesHomogeneous = types.map { it.dialectType to it.javaType }.distinct().size == 1
```

Comment:
Is it worth making this lazy or nesting it under the check for `preferKotlinType`?

Reply:
I don't think so.